rpc/grpc/server/internal/interceptors/unary/unary.go

```diff
@@ -86,28 +86,39 @@ func (u *Interceptor) Intercept(ctx context.Context, req any, info *grpc.UnarySe
 	ctx, spanner := span.New(ctx, opts...)
 	defer spanner.End()
 
+	var err error
+	for _, i := range u.intercepts {
+		req, err = i(ctx, req, grpcMeta)
+		if err != nil {
+			return nil, u.errLogAndConvert(ctx, err, grpcMeta, req)
+		}
+	}
+
 	resp, err := handler(ctx, req)
 	if err != nil {
-		if e, ok := err.(errors.Error); ok {
-			e.Log(ctx, grpcMeta.CallID, grpcMeta.CustomerID, req)
-			if u.errConvert != nil {
-				status, err := u.errConvert(ctx, e, grpcMeta)
-				if err != nil {
-					return nil, err
-				}
-				return nil, status.Err()
-			}
-			return nil, e
-		} else {
-			e := errors.E(ctx, nil, nil, err)
-			e.Log(ctx, grpcMeta.CallID, grpcMeta.CustomerID, req)
-			return nil, e
-		}
+		return nil, u.errLogAndConvert(ctx, err, grpcMeta, req)
 	}
 
 	return resp, err
 }
 
+func (u *Interceptor) errLogAndConvert(ctx context.Context, err error, grpcMeta grpcContext.Metadata, req any) error {
+	if e, ok := err.(errors.Error); ok {
+		e.Log(ctx, grpcMeta.CallID, grpcMeta.CustomerID, req)
+		if u.errConvert != nil {
+			status, err := u.errConvert(ctx, e, grpcMeta)
```

Comment:
Think this is going to not work as excected.

Ok, so if u.errConvert returns an error, you can log that error if you want.  But if you assign it to "err", you are creating a shadow variable of "err" that erases the passed "err" that comes in as the argument.  That isn't much of a problem, because you've converted the error to "e".   But what is a problem is you return "err" on errConvert.  Your returning the conversion error.  Might be worth logging it, but not returning it.  You need to return "e" if errConvert doesn't work.  Either way you also need to log "e".

Reply:
Nevermind about the log part for e.

Reply:
`e` is already logged here. But returning e here does sound reasonable?

Reply:
No, because you'd be returning ce, which gets pushed out to the client.  But its a logging conversion error so it shouldn't go to the user.  You can log it via context.Log(), but don't convert it to errors.Error().   

```go
status, cErr := u.errConvert(ctx, e, grpcMeta)
if cErr != nil {
  context.Log(ctx).Info(cErr.Error())
  e.Log(ctx, ctx, grpcMeta.CallID, grpcMeta.CustomerID, req)
  return e
}
...
```

Reply:
whoops yeah I think that's what I meant originally anyway